### DISCUSSION TITLE
How to handle lunar based holidays?

### DISCUSSION THREAD

**Discussion by Nielsvanpach** [spatie/holidays#79] [Ideas]
I'm open to PRs on this!

**Comment by Nielsvanpach** [comment]
Could this repository help? https://github.com/GeniusTS/hijri-dates

**Reply by AvishkaSen** [reply]
https://github.com/BitAndBlack/php-moon-phase
You could use this too 😃 

I forked the Spatie holidays repo and worked on adding Sri Lankan holidays. 
I calculated the full moon holidays (poya) holidays with that repo.

And it works with that repo. 😄 
Planning on submitting a PR tomorrow but there are some issues with how the holidays are calculated here. 
But the above repo works great.

**Comment by kohenkatz** [comment]
There are multiple lunar calendars that exist. How to handle lunar holidays will probably depend which lunar calendar you are referring to.

PHP has built-in support for many of them (including Chinese, Buddhist, Indian, Islamic, Hebrew, Persian, and others), though documentation is severely lacking.

You already have a repo linked above for Islamic dates, and you already have at least a partial implementation of the Chinese lunar calendar that is used for Indonesia and Vietnam (committed after this discussion was started).

Is there something specific you are looking for that might be different from how the Chinese implementation has already been started?

**Reply by Nielsvanpach** [reply]
The package i linked to above hasn't received updates in more than 3y, so we're not using that one.
Also, as stated for Turkey, the generated results aren't correct: https://github.com/spatie/holidays/blob/main/src/Countries/Turkey.php#L11-L17
